src/v/test_utils/randoms.h

```diff
@@ -39,4 +46,97 @@ random_broker(int32_t id_low_bound, int32_t id_upper_bound) {
         .cores = random_generators::get_int<uint32_t>(96)});
 }
 
-} // namespace tests
\ No newline at end of file
+inline bool random_bool() { return random_generators::get_int(0, 100) > 50; }
+
+template<typename T>
+T random_choice(std::vector<T> elements) {
+    auto idx = random_generators::get_int<size_t>(0, elements.size() - 1);
+    return elements[idx];
+}
+
+template<typename T>
```

Comment:
I have a change in review that adds an interface similar to what `rapidcheck` [does](https://github.com/emil-e/rapidcheck).
```
template<typename T>
struct gen {};

template<typename T>
std::vector<T> gen_array(size_t n) { 
    std::vector<T> v;
    v.reserve(n);
    for(auto i=0; i < n; ++i) {
        v[i] = gen<T>::generate();
    }
    return v;
}
```
Tests can implement their own generators:
```
template<>
struct gen<NamedInteger> {
    static NamedInteger generate() {
        return NamedInteger{random_generators::gen_int()};
    }
};
````  
Then you can use it in a generic way.
```
auto v = gen_array<NamedInteger>();
````